src/main/java/org/reficio/p2/P2Mojo.java

```diff
@@ -205,7 +227,7 @@ public void execute() {
         try {
             initializeEnvironment();
             initializeRepositorySystem();
-            processArtifacts();
+            processArtifacts(this.artifacts);
```

Comment:
This method is used also in createFeature() to process the artifacts of the feature definitions, which may differ.